app/controllers/surveys_controller.rb

```diff
@@ -46,15 +40,15 @@ def create
     end
   end
 
+  # edit surveys/:id/edit
   def edit
-    @survey = Survey.find(params[:id])
     respond_to do |format|
       format.html
     end
   end
 
+  # patch surveys/:id
   def update
-    @survey = Survey.find(params[:id])
     if @survey.update(survey_params)
       flash[:notice] = 'Survey Updated!'
       redirect_to @survey
```

Comment:
show @survey.errors.full_messages in else case instead of 'Not Updated' message
do the same in every action